src/wasm.h

```diff
@@ -1471,6 +1476,32 @@ class Try : public SpecificExpression<Expression::TryId> {
   void finalize(Type type_);
 };
 
+// 'try_table' from the new EH proposal
+class TryTable : public SpecificExpression<Expression::TryTableId> {
+public:
+  TryTable(MixedArena& allocator)
+    : catchTags(allocator), catchDests(allocator), catchRefs(allocator),
+      sentTypes(allocator) {}
+
+  Expression* body;
+
+  // Tag names. Empty names (Name()) for catch_all and catch_all_ref
+  ArenaVector<Name> catchTags;
+  // catches' destination blocks
+  ArenaVector<Name> catchDests;
+  // true for catch_ref and catch_all_ref
+  ArenaVector<bool> catchRefs;
+
+  bool hasCatchAll() const;
+
+  void finalize(Module* wasm = nullptr);
```

Comment:
Please add a comment here about the optional Module.

Reply:
Done